components/inputs/CodeInput.tsx

```diff
@@ -40,44 +41,50 @@ type CodeInputProps = Omit<
  * @param cellCount
  * @param value
  * @param setValue
+ * @param error string that will change border color to negative and show error message
  * @param props
  * @constructor
  */
-const CodeInput = ({ cellCount = 6, value, setValue, ...props }: CodeInputProps) => {
+const CodeInput = ({ cellCount = 6, value, setValue, error, ...props }: CodeInputProps) => {
   const ref = useBlurOnFulfill({ value, cellCount })
   const [focusCellProps, getCellOnLayoutHandler] = useClearByFocusCell({
     value,
     setValue,
   })
 
   return (
-    <CodeField
-      ref={ref}
-      {...props}
-      {...focusCellProps}
-      // Use `caretHidden={false}` when users can't paste a text value, because context menu doesn't appear
-      value={value}
-      onChangeText={setValue}
-      cellCount={cellCount}
-      keyboardType="number-pad"
-      autoComplete="one-time-code"
-      rootStyle={styles.rootStyle}
-      renderCell={({ index, symbol, isFocused }) => (
-        <Typography
-          key={index}
-          onLayout={getCellOnLayoutHandler(index)}
-          // TODO - sizes are little bit guessed, needs to be tested on multiple devices
-          className={clsx(
-            'h-[48px] w-[48px] items-center justify-center rounded border border-divider bg-white text-center text-[20px] leading-[44px]',
-            {
-              'border-dark': isFocused,
-            },
-          )}
-        >
-          {symbol || (isFocused ? <Cursor /> : null)}
-        </Typography>
-      )}
-    />
+    <View className="flex-col g-1">
+      <CodeField
+        ref={ref}
+        {...props}
+        {...focusCellProps}
+        // Use `caretHidden={false}` when users can't paste a text value, because context menu doesn't appear
+        value={value}
+        onChangeText={setValue}
+        cellCount={cellCount}
+        keyboardType="number-pad"
+        autoComplete="one-time-code"
+        rootStyle={styles.rootStyle}
+        renderCell={({ index, symbol, isFocused }) => (
+          <Typography
+            key={index}
+            onLayout={getCellOnLayoutHandler(index)}
+            // TODO - sizes are little bit guessed, needs to be tested on multiple devices
+            className={clsx(
+              'h-[48px] w-[48px] items-center justify-center rounded border border-divider bg-white text-center text-[20px] leading-[44px]',
+              {
+                'border-dark': isFocused,
+                'border-negative': !!error,
+              },
+            )}
+          >
+            {symbol || (isFocused ? <Cursor /> : null)}
+          </Typography>
+        )}
+      />
+
+      {error ? <Typography className="text-negative">{error}</Typography> : null}
```

Comment:
If `react-native-confirmation-code-field` does not have an error props, than okay.
But we should keep in mind also accessibility.

Reply:
There is no single line about errors in that lib... I even tried to make it myself, but as I read in react-native docs there is no such thing as `aria-invalid` or any similar accessibility field...